backend/rust-bindings/src/lib.rs

```diff
@@ -41,26 +60,11 @@ pub extern "C" fn schema_display(
     schema: *const c_char,
     schema_version: FFIOption,
     result: *mut *mut c_char) -> bool {
-    
-    let schema_hex = match get_str_from_pointer(schema, result) {
-        Ok(out) => out,
-        Err(_) => return false,
-    };
 
-    let display = match schema_display_aux(schema_hex, schema_version.into_option()) {
-        Ok(display) => display,
-        Err(e) => {
-            unsafe {
-                *result = CString::new(e.to_string()).unwrap().into_raw()    
-            }
-            return false
-        }
-    };
-    unsafe {
-        *result = CString::new(display.to_string()).unwrap().into_raw()
-    }
-
-    true
+    assign_result(result, (|| {
```

Comment:
IMHO this doesn’t look more clean. Here a closure is defined which is executed at once.

Passing closures around is something you normally see - ex. every time you use `map`.

Reply:
The solution I had in mind was more like
```
let schema_hex = get_str_from_pointer(schema)?;
let schema_res = schema_display_aux(schema_hex, schema_version.into_option());
assign_result(result, schema_res)
```

Reply:
Well that wouldn't deal with the results from the `get_str_from_pointer`.
You need to have a closure (or move this stuff into separate functions) to avoid dealing with all the results individually, like was done originally.

Reply:
makes sense